src/representer/mapping.nim

```diff
@@ -0,0 +1,45 @@
+import hashes, tables, strutils
+
+type
+  NormalizedIdent = string
+  IdentMap* = OrderedTable[NormalizedIdent, string]
+
+proc hash(x: NormalizedIdent): Hash {.used.} =
+  !$(x[0].hash !& x[1..^1].hashIgnoreStyle)
+
+proc `==`(a, b: NormalizedIdent): bool =
+  a[0] == b[0] and
+    a.replace("_", "").toLowerAscii == b.replace("_", "").toLowerAscii
```

Comment:
Here there's `==` between two strings, within a proc that declares `==` on two `NormalizedIdent` (which are strings). So running `nim c -r mapping.nim` just errors when it hits the recursion limit. 

But I don't know what this proc is for - it isn't currently used. And to me, the name "NormalizedIdent" implies that they're already normalized, so the `==` shouldn't do normalization? The name sounds like it should be normalized on creation.

Should it just be a `distinct string`?

Alternatively:
1. `macros.eqIdent`?
1. Compare without using the string equality operator? The below uses only `!=` on chars, and uses `strutils.nimIdentNormalize` instead.
```suggestion
  let aNorm = nimIdentNormalize(a)
  let bNorm = nimIdentNormalize(b)
  if aNorm.len != bNorm.len:
    return false
  for i in 0 ..< aNorm.len:
    if aNorm[i] != bNorm[i]:
      return false
  result = true
```

Reply:
> So running nim c -r mapping.nim just errors when it hits the recursion limit.

Fixed it. Forgot to commit that change and it got removed 🤷‍♂️ 

> But I don't know what this proc is for - it isn't currently used. And to me, the name "NormalizedIdent" implies that they're already normalized, so the == shouldn't do normalization? The name sounds like it should be normalized on creation.

What I meant by normalized is in the source code, references to this ident are normalized. Push a commit to fix it.

>Should it just be a `distinct string`?

I wasn't sure. I only care about `helloWorld == hello_world` `Dollar != dollar` in the `Table` which is ensured (I think/hope) by just implementing those two procs as outlined in the docs [here](https://nim-lang.org/docs/tables.html#basic-usage-hashing). I do not care whether it is `distinct` or not. I just thought that it would be better to be able to use all the `string` procs and fund

> Alternatively:
> 1. `macros.eqIdent`?
> 1. Compare without using the string equality operator? The below uses only `!=` on chars, and uses `strutils.nimIdentNormalize` instead.

1. Doesn't work at runtime, so we can't compare the `IdentMap` at runtime
2. Ironically it doesn't exist in 1.2.x

Reply:
Turns out my solution didn't work outside of the module (I had only tested it with `when isMainModule` & `doAssert`) so I changed to use `distinct string` and implemented the relevant procs necessary.

It working in testing

Reply:
> 1. Doesn't work at runtime, so we can't compare the IdentMap at runtime

True. But is the intention that the representer works entirely at compile-time, and the run-time context is just for tests?

> 2. Ironically it doesn't exist in 1.2.x

Ah, good point. Sorry, I tend to live on `devel`. Turns out `strutils.nimIdentNormalize` was a few weeks too late to make it into 1.2.

> I pushed a commit to fix it which uses `cmpIgnoreStyle`

Nice. "Check the first char is identical, then use `cmpIgnoreStyle`" is much better than using `nimIdentNormalize` anyway - the latter route is more verbose and allocates more.

Reply:
> True. But is the intention that the representer works entirely at compile-time, and the run-time context is just for tests?

To use a "compile-time only" comparison would require lots of wrangling to use `unittest`, especially for "expected vs. actual" and line information